crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2289,7 +2289,16 @@ impl<'db> TypeInferenceBuilder<'db> {
                                 // (`Type::format`?) that handles the `__format__` method.
                                 // Conversion flags should be handled before calling `__format__`.
                                 // https://docs.python.org/3/library/string.html#format-string-syntax
-                                if !conversion.is_none() || format_spec.is_some() {
+                                if !conversion.is_none() {
```

Comment:
```suggestion
                                if conversion.is_some() {
```

Reply:
`conversion` is a `ConversionFlag` enum with a `None` element, not an `Option`